model/fn-execution/src/main/proto/metric_definitions.yaml

```diff
@@ -0,0 +1,57 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Metrics Definitions describing various BEAM metrics.
+# See: https://s.apache.org/beam-fn-api-metrics
+
+- annotations:
+    description: The total estimated execution time of the ptransform
+    unit: msecs
+  labels:
+  - PTRANSFORM
+  type: beam:metrics:SumInt64
+  urn: beam:metric:ptransform_execution_time:total_msecs:v1
+- annotations:
+    description: The total estimated execution time of the start bundle function in
+      a pardo
+    unit: msecs
+  labels:
+  - PTRANSFORM
+  type: beam:metrics:SumInt64
+  urn: beam:metric:pardo_execution_time:start_bundle_msecs:v1
+- annotations:
+    description: The total estimated execution time of the process bundle function
+      in a pardo
+    unit: msecs
+  labels:
+  - PTRANSFORM
+  type: beam:metrics:SumInt64
+  urn: beam:metric:pardo_execution_time:process_bundle_msecs:v1
+- annotations:
+    description: The total estimated execution time of the finish bundle function
+      in a pardo
+    unit: msecs
+  labels:
+  - PTRANSFORM
+  type: beam:metrics:SumInt64
+  urn: beam:metric:pardo_execution_time:finish_bundle_msecs:v1
+- annotations:
+    description: The total elements counted for a metric.
+  labels:
+  - PTRANSFORM
+  type: beam:metrics:SumInt64
+  urn: beam:metric:element_count:v1
```

Comment:
Can you add examples of other metrics? e.g. distribution / gauge types

Reply:
This file is for system defined ones, which we don't seem to have. As they are only used in use counters. Unless I am missing something. Happy to add one if we have an example in another SDK perhaps.

Right now this is just a catalog, its not actually loaded or used anywhere.

The doc covers how to add more.